packages/components/src/Filter/FilterBar/context/reducer/filterBarStateReducer.ts

```diff
@@ -46,5 +47,19 @@ export const filterBarStateReducer = <ValuesMap extends FiltersValues>(
         ...updateDependentFilters(state),
         hasUpdatedValues: true,
       }
+
+    case "update_filter_labels":
+      return {
+        ...state,
+        filters: Object.values(state.filters).reduce((acc, filter) => {
+          acc[filter.id] = {
+            ...filter,
+            name:
+              action.data.find(({ id }) => id === filter.id)?.name ??
+              filter.name,
+          }
+          return acc
```

Comment:
To prevent the O(n^2) complexity here, I can create a keyed object for the name lookup. Because it's unlikely this data will be large, I didn't bother.